cedar-lean/Cedar/Thm/Validation/Typechecker/BinaryApp.lean

```diff
@@ -807,16 +815,15 @@ theorem type_of_mem_is_sound {x₁ x₂ : Expr} {c₁ c₂ : Capabilities} {env
   GuardedCapabilitiesInvariant (Expr.binaryApp .mem x₁ x₂) c₂ request entities ∧
   ∃ v, EvaluatesTo (Expr.binaryApp .mem x₁ x₂) request entities v ∧ InstanceOfType v ty
 := by
-  rcases (type_of_mem_inversion h₃) with ⟨hc, ety₁, ety₂, ⟨c₁', h₄⟩ , c₂', h₅⟩
+  have ⟨hc, ety₁, ety₂, ⟨c₁', h₄⟩ , c₂', h₅⟩ := type_of_mem_inversion h₃
   subst hc
   constructor
   case left => exact empty_guarded_capabilities_invariant
   case right =>
-    rcases h₅ with h₅ | h₅ <;>
-    rcases h₅ with ⟨h₅, h₆⟩ <;> subst h₆
-    case inl.intro =>
+    rcases h₅ with ⟨h₅, h₆⟩ | ⟨h₅, h₆⟩ <;> subst h₆
```

Comment:
Not sure what the `|` syntax is here. Is this a case where `rcases` can't be replaced due to recursion?